src/main/scala/com/fulcrumgenomics/bam/Bams.scala

```diff
@@ -201,8 +199,8 @@ object Bams extends LazyLogging {
     }
     else {
       val refBases = ref.get(rec.getReferenceIndex).getBases
-      SequenceUtil.calculateMdAndNmTags(rec, refBases, rec.getAttribute(MD.name()) != null, rec.getAttribute(NM.name()) != null)
-      if (rec.getAttribute(SAMTag.UQ.name) != null && rec.getBaseQualities != null && rec.getBaseQualities.length != 0) {
+      SequenceUtil.calculateMdAndNmTags(rec, refBases, true, true)
```

Comment:
I realized I couldn't do the "replace only if it's there strategy" as we decided that the clipping code should forcefully remove tags that are invalid ... so I need to fill them in on reads that don't have them.  Oh well.